src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,111 @@
+    @Override
+    public Optional<Product> getById(Long id) {
+        String query = "SELECT * FROM products WHERE deleted = false AND product_id = ?";
+        Product product = new Product(null, 0);
+        try (Connection connection = ConnectionUtil.getConnection();) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                return Optional.ofNullable(getProductFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to get product with ID " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products WHERE deleted = FALSE;";
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Product product = getProductFromResultSet(resultSet);
+                products.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all products", e);
+        }
+        return products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET product_name = ?, price = ? "
+                + "WHERE deleted = FALSE AND product_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.setLong(3, product.getId());
+            statement.executeUpdate();
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update product with ID "
+                    + product.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE products SET deleted = TRUE WHERE product_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete product with ID " + id, e);
+        }
+    }
+
+    public Product getProductFromResultSet(ResultSet resultSet)
+            throws SQLException {
+        long productId = resultSet.getLong("product_id");
+        String name = resultSet.getString("product_name");
```

Comment:
```suggestion
        String name = resultSet.getString("name");
```

Reply:
column got name "product_name", so i can`t

Reply:
so what is the problem with changing name of the column in the database ?